src/cluster_legacy.c

```diff
@@ -186,6 +187,49 @@ dictType clusterSdsToListType = {
     NULL                /* allow to expand */
 };
 
+typedef struct {
+    enum { ITER_DICT, ITER_LIST } type;
+    union {
+        dictIterator di;
+        listIter li;
+    };
+} ClusterNodeIterator;
+
+void clusterNodeIterInitAllNodes(ClusterNodeIterator *iter, dict *d) {
+    iter->type = ITER_DICT;
+    dictInitSafeIterator(&iter->di, d);
+}
+
+void clusterNodeIterInitMyShard(ClusterNodeIterator *iter, list *l) {
+    iter->type = ITER_LIST;
+    listRewind(l, &iter->li);
```

Comment:
Let's do the actual "my shard" logic here.

```suggestion
void clusterNodeIterInitMyShard(ClusterNodeIterator *iter) {
    list *nodes_for_slot = clusterGetNodesInMyShard(server.cluster->myself);
    serverAssert(nodes_for_slot != NULL);
    iter->type = ITER_LIST;
    listRewind(nodes_for_slot, &iter->li);
```